cpp/src/io/comp/nvcomp_adapter.hpp

```diff
@@ -30,14 +32,52 @@ namespace cudf::io::nvcomp {
 enum class compression_type { SNAPPY, ZSTD, DEFLATE };
 
 /**
- * @brief Whether the given compression type is enabled through nvCOMP.
+ * @brief Set of parameters that impact whether the use nvCOMP features is enabled.
+ */
+struct feature_status_parameters {
+  int lib_major_version;
+  int lib_minor_version;
+  int lib_patch_version;
+  bool are_all_integrations_enabled;
+  bool are_stable_integrations_enabled;
+  int compute_capability;
+
+  feature_status_parameters();
+  feature_status_parameters(
+    int major, int minor, int patch, bool all_enabled, bool stable_enabled, int cc)
```

Comment:
Compute capabilities are not really integral, there are lots of values like `8.6` or `6.1`. Does this get some kind of normalized integral value like `86`? How does this interact with `params.compute_capability == 6`?

Reply:
Looks like my naming is bad here. 
This is just the major capability number.
`cudaDeviceGetAttribute(&compute_capability, cudaDevAttrComputeCapabilityMajor, device))` returns an int so this should be fine. 
I'll fix the name to avoid confusion.

Reply:
Renamed to reflect that this is the `major` component of compute capability.